### ISSUE TITLE
Operator does not cleanup secret

### ISSUE THREAD

**Issue by mgagliardo91** [authzed/spicedb-operator#99]
We are currently using the operator with a generated `SpiceDBCluster` CRD. When deleting the CRD, the operator successfully deletes all related entities (services, pods, etc); however, it leaves behind the secret used to store the shared secret key. This requires us to manually delete the secret before we can reinstall.

**Comment by ecordell** [comment]
The secret is provided as input to the SpiceDBCluster - it isn't deleted because it's not "owned" by the cluster. 

How are you creating the secret and cluster? Can you just remove the secret at the same time you remove the cluster?

**Comment by mgagliardo91** [comment]
@ecordell that was my expectation as well, but even though I am creating the secret, the operator appears to be putting an annotation and a label on the secret once the `SpiceDBCluster` is applied, indicating that it is owned by the cluster:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: staging-nio-permissions-config
  namespace: spicedb-play
  labels:
    authzed.com/managed-by: operator
  annotations:
    authzed.com.cluster-owner/staging-nio-permissions: owned
  managedFields:
    - manager: spicedb-operator
      operation: Apply
      apiVersion: v1
      time: '2022-10-24T20:27:03Z'
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:labels:
            f:authzed.com/managed-by: {}
    - manager: spicedbcluster-owner-spicedb-play-staging-nio-permissions
      operation: Apply
      apiVersion: v1
      time: '2022-10-24T20:27:03Z'
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            f:authzed.com.cluster-owner/staging-nio-permissions: {}
type: Opaque
data:
   <removed>
```

These annotations are enough that the secret does not get removed by our helm chart that generates the `SpiceDBCluster` CRD and other relevant items.

**Comment by mgagliardo91** [comment]
@ecordell currently on Helm v3.8.2. Currently, our helm chart just creates a secret and the associated `SpiceDBCluster` CRD. I would expect that a `helm uninstall` would remove both the cluster CRD (which it does) as well as the secret it created. We definitely have other helm charts that create/remove secrets correctly, so I was wondering if it was related to the extra labels/annotations when helm does its lookup, but I'm not quite sure. I can send a stripped down version of the chart if that would be helpful.

**Comment by ecordell** [comment]
I tried to reproduce with helm 3.10.1:

- created a helm chart with a `SpiceDBCluster` and `Secret`
- installed the chart with `helm install`
- confirmed the cluster ran and the secret had spicedb annotations
- ran `helm uninstall` and saw it remove both the cluster and the secret

Maybe try updating your version of helm? Or do you know if there are other settings you're using when installing/uninstalling that might affect this behavior?

**Comment by mgagliardo91** [comment]
@ecordell I installed with a different version and didn't run into the same problem. Thanks! I have a separate question, but I'll open a new issue for it.
